### PR TITLE
Issue 156 - Added payment confirmation screen

### PR DESCRIPTION
**Pull request by ahmed-shehata**
Added payment confirmation screen (tested on Paypal)

### CODE REVIEW

**Comment by ahmed-shehata** [comment]
:+1:

File: AtlasUI/AtlasUI/PaymentProcessingViewController.swift [hunk 1]
```diff
@@ -45,7 +45,23 @@ internal final class PaymentProcessingViewController: UIViewController, Checkout
...
+                }
+                let paymentSelectionViewController = PaymentSelectionViewController(paymentSelectionURL: paymentURL)
+                paymentSelectionViewController.paymentCompletion = { _ in
+                    self.showSuccessImage()
+                }
+
+                guard let sourceApp = UIApplication.topViewController() else { return }
+
+                let navigationController = UINavigationController(rootViewController: paymentSelectionViewController)
+                dispatch_async(dispatch_get_main_queue()) {
```
Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/PaymentProcessingViewController.swift]: `Async.main` ?

File: AtlasUI/AtlasUI/PaymentProcessingViewController.swift [hunk 2]
```diff
@@ -45,7 +45,23 @@ internal final class PaymentProcessingViewController: UIViewController, Checkout
...
+                    navigationController.modalPresentationStyle = .OverCurrentContext
+                    sourceApp.navigationController?.presentViewController(navigationController, animated: true, completion: nil)
```
Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/PaymentProcessingViewController.swift]: what if the sourceApp doesn't have a navigationController?

Comment (ahmed-shehata) [inline AtlasUI/AtlasUI/PaymentProcessingViewController.swift]: Then it should return

Comment (ahmed-shehata) [inline AtlasUI/AtlasUI/PaymentProcessingViewController.swift]: "?" optional

**Comment by ahmed-shehata** [comment]
👍

File: AtlasUI/AtlasUI/PaymentProcessingViewController.swift (hunk 2 above)
Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/PaymentProcessingViewController.swift]: Yeah I know, I mean if the client doesn't have a navigationController then this view will not be displayed... Which is not good I think :(

**Comment by ahmed-shehata** [comment]
👍

**Comment by hani-ibrahim** [comment]
👍

**Pull request merged**